src/retry.decorator.ts

```diff
@@ -22,36 +22,30 @@ export function Retryable(options: RetryOptions): DecoratorFunction {
       setExponentialBackOffPolicyDefault();
     }
     descriptor.value = async function(...args: any[]) {
-      try {
-        return await retryAsync.apply(this, [originalFn, args, options.maxAttempts, options.backOff]);
-      } catch (e) {
-        if (e instanceof MaxAttemptsError) {
-          const msgPrefix = `Failed for '${propertyKey}' for ${options.maxAttempts} times.`;
-          e.message = e.message ? `${msgPrefix} Original Error: ${e.message}` : msgPrefix;
-        }
-        throw e;
-      }
+      return await retryAsync.apply(this, [originalFn, args, options.maxAttempts, options.backOff]);
     };
     return descriptor;
   };
 
   async function retryAsync(fn: () => any, args: any[], maxAttempts: number, backOff?: number): Promise<any> {
-    try {
-      return await fn.apply(this, args);
-    } catch (e) {
-      if (--maxAttempts < 0) {
-        e?.message && console.error(e.message);
-        throw new MaxAttemptsError(e?.message);
-      }
-      if (!canRetry(e)) {
-        throw e;
-      }
-      backOff && (await sleep(backOff));
-      if (options.backOffPolicy === BackOffPolicy.ExponentialBackOffPolicy) {
-        const newBackOff: number = backOff * options.exponentialOption.multiplier;
-        backOff = newBackOff > options.exponentialOption.maxInterval ? options.exponentialOption.maxInterval : newBackOff;
-      }
-      return retryAsync.apply(this, [fn, args, maxAttempts, backOff]);
+    for (let i = 0; i<(maxAttempts+1); i++) {
+      try {
+        return await fn.apply(this, args);
+      } catch (e) {
+        if (i == maxAttempts) {
+          if (options.reraise) {
```

Comment:
can we rename this option as something like `useOriginalError` to reduce confusion?